src/rust/src/address_ffi.rs

```diff
@@ -72,8 +72,8 @@ impl UnifiedAddressHelper {
         }
 
         self.ua
-            .items_as_parsed()
-            .iter()
+            .items()
```

Comment:
This is reverting a bugfix from #5429 (which had never made it over to the wallet branch). However, #5450 will be changing this logic to the now-canonical encoding order, so it's okay to leave this as-is.